src/dopri.ts

```diff
@@ -0,0 +1,149 @@
+import * as dopri5 from "./dopri5/stepper";
+import * as interpolator from "./interpolator";
+import {RhsFn, Stepper} from "./types";
+import * as utils from "./utils";
+
+// needed for ES5 - will be ~= Number.EPSILON in ES6
+const DBL_EPSILON = Math.pow(2, -52); // = 2.220446049250313e-16
+const STEP_FACTOR_MIN = 1e-4;
+
+export function integrate(rhs: RhsFn, y: number[],
+                          t0: number, t1: number) {
+    const solver = new Dopri(rhs, y.length);
+    solver.initialise(t0, y);
+    return solver.run(t1);
+}
+
+function integrationError(message: string, t: number) {
+    return new Error(`Integration failure: ${message} at ${t}`);
```

Comment:
This is kind of pedantic but I think this is more naturally a class than a function
```
 class IntegrationError extends Error {
        constructor(message:string, t: number) {
            super(`Integration failure: ${message} at ${t}`);
        }
    }
```
then
```
throw new IntegrationError("too many steps", t)
```

Reply:
Does this work! I googled around for extending error all over the show and it seemed to be a mess! Things like [this](https://stackoverflow.com/questions/41102060/typescript-extending-error-class) and [this](https://www.metachris.com/2017/01/custom-errors-in-typescript-2.1/) which I got totally confused about

Reply:
here's the official comments on this: https://github.com/Microsoft/TypeScript-wiki/blob/master/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work - I've had limited success in getting this working...

Reply:
oh, my bad, didn't know this was so fraught! Stick to your way then